### PR TITLE
Implement GM2+ styles

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue:

- #5254 

[Storybook Preview](https://google.github.io/site-kit-wp/storybook/pull/5314/)

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Comment by felixarntz** [comment]
For reference, posting the [Slack thread with feedback on the UI](https://10up.slack.com/archives/C01CA828ZPZ/p1654690914461999) here.

File: assets/js/components/settings/SettingsActiveModules.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default function SettingsActiveModules() {
 		.sort( ( a, b ) => a.order - b.order );
 
 	return (
-		<Layout>
+		<Layout className="googlesitekit-layout--rounded">
```
Comment (felixarntz) [inline assets/js/components/settings/SettingsActiveModules.js:43]: It may be a bit cleaner to add this class via a prop, e.g. a `rounded` boolean prop.

File: assets/js/components/settings/SettingsAdmin.js [hunk 2]
```diff
@@ -82,7 +82,7 @@ export default function SettingsAdmin() {
 			{ userInputEnabled && (
 				<Cell size={ 12 }>
 					{ isUserInputCompleted && (
-						<Layout>
+						<Layout className="googlesitekit-layout--rounded">
```
Comment (felixarntz) [inline assets/js/components/settings/SettingsAdmin.js:85]: See above.

File: assets/js/components/settings/SettingsAdmin.js [hunk 3]
```diff
@@ -108,13 +108,16 @@ export default function SettingsAdmin() {
 						</Layout>
 					) }
 					{ ! isUserInputCompleted && (
-						<UserInputSettings isDismissible={ false } />
+						<UserInputSettings
+							isDismissible={ false }
+							className="googlesitekit-publisher-win--rounded"
```
Comment (felixarntz) [inline assets/js/components/settings/SettingsAdmin.js:113]: Same here.

File: assets/js/components/settings/SettingsAdmin.js [hunk 4]
```diff
@@ -108,13 +108,16 @@ export default function SettingsAdmin() {
...
+						/>
 					) }
 				</Cell>
 			) }
 
 			<Cell size={ 12 }>
-				<Layout>
+				<Layout className="googlesitekit-layout--rounded">
```
Comment (felixarntz) [inline assets/js/components/settings/SettingsAdmin.js:120]: Same here and throughout similar adjustments in the PR.

File: assets/js/components/settings/SettingsApp.js [hunk 5]
```diff
@@ -70,9 +70,10 @@ function SettingsApp() {
 							/>
 						</Cell>
 						<Cell size={ 12 }>
-							<Layout>
+							<Layout className="googlesitekit-layout--rounded googlesitekit-layout--transparent">
```
Comment (felixarntz) [inline assets/js/components/settings/SettingsApp.js:73]: Same here, maybe another prop for the transparent bit would help.

File: assets/js/components/DetailsPermaLinks.js [hunk 6]
```diff
@@ -49,7 +49,11 @@ export default function DetailsPermaLinks( { title, path, serviceURL } ) {
 
 	return (
 		<Fragment>
-			<Link href={ serviceURL || detailsURL } external={ !! serviceURL }>
+			<Link
+				href={ serviceURL || detailsURL }
+				external={ !! serviceURL }
+				hideExternalIndicator
```
Comment (felixarntz) [inline assets/js/components/DetailsPermaLinks.js:55]: I'm not sure I'm onboard with this change as it may be problematic for accessibility reasons? Thoughts @tofumatt?

Marking external links as such is an a11y best practice, and I can't imagine that Material 3 would dictate otherwise.

File: assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidget.js [hunk 7]
```diff
@@ -180,7 +180,14 @@ function DashboardTopEarningPagesWidget( props ) {
 			primary: true,
 			Component: ( { row } ) => {
 				const [ title, url ] = row.dimensions;
-				return <Link href={ url } children={ title } external />;
+				return (
+					<Link
+						href={ url }
+						children={ title }
+						external
+						hideExternalIndicator
```
Comment (felixarntz) [inline assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidget.js:188]: See above - why was this changed, could we remove this prop again?

File: assets/js/modules/search-console/components/dashboard/DashboardPopularKeywordsWidget.js [hunk 8]
```diff
@@ -155,7 +155,11 @@ function DashboardPopularKeywordsWidget( props ) {
 				} );
 
 				return (
-					<Link href={ searchAnalyticsURL } external>
+					<Link
+						href={ searchAnalyticsURL }
+						external
+						hideExternalIndicator
```
Comment (felixarntz) [inline assets/js/modules/search-console/components/dashboard/DashboardPopularKeywordsWidget.js:161]: Same here.

**Review by felixarntz — APPROVED** [review]
@asvinb Excellent work!

I left a few comments on the JS, but nothing blocking. I didn't review the CSS changes in depth since I don't think that's needed here, especially given that we're not going to keep the current rather messy CSS approach for very much longer. Tested this PR and looked at different areas, all looks solid to me now. 🎉

File: assets/js/components/settings/SettingsActiveModules.js (hunk 1 above)
Comment (asvinb) [reply assets/js/components/settings/SettingsActiveModules.js]: 👍

File: assets/js/components/settings/SettingsAdmin.js (hunk 3 above)
Comment (asvinb) [reply assets/js/components/settings/SettingsAdmin.js]: 👍

File: assets/js/components/settings/SettingsApp.js (hunk 5 above)
Comment (asvinb) [reply assets/js/components/settings/SettingsApp.js]: 👍

File: assets/js/components/DetailsPermaLinks.js (hunk 6 above)
Comment (asvinb) [reply assets/js/components/DetailsPermaLinks.js:55]: @felixarntz We are only hiding the icon for some external links. I don't think it'll cause any accessibility issues since the `aria-label` states that the link will open in a new tab. Hiding the external icon is something that UX requested for links within tables.

File: assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidget.js (hunk 7 above)
Comment (asvinb) [reply assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidget.js:188]: @felixarntz See my comment here: https://github.com/google/site-kit-wp/pull/5314#discussion_r907369159

File: assets/js/modules/search-console/components/dashboard/DashboardPopularKeywordsWidget.js (hunk 8 above)
Comment (asvinb) [reply assets/js/modules/search-console/components/dashboard/DashboardPopularKeywordsWidget.js:161]: @felixarntz See my comment here: https://github.com/google/site-kit-wp/pull/5314#discussion_r907369159

File: assets/sass/config/_variables-mui3.scss [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+$fs-display-sm: 2.375rem; // 38px
+$fs-headline-lg: 2rem; // 32px
+$fs-headline-md: 1.75rem; // 28px
+$fs-headline-sm: 1.5rem; // 24px
+$fs-title-lg: 1.25rem; // 20px
+$fs-title-md: 1rem; // 16px
+$fs-title-sm: 0.875rem; // 14px
+$fs-body-lg: 1rem; // 16px
+$fs-body-md: 0.875rem; // 14px
+$fs-body-sm: 0.75rem; // 12px
```
Comment (aaemnnosttv) [inline assets/sass/config/_variables-mui3.scss:184]: Why not use the `px` values? This is particularly important for the frontend in the admin bar where the base font size `rem` are relative to is outside of our control and so we can't rely on these to map to any specific absolute values. See #888

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @asvinb, great work! This looks very close overall, there are a few things I left comments on, otherwise I have a few other comments to mention:

## Green hues

I find the mix of greens a bit off looking.

The first example has two different colors on clickable text which feels a bit inconsistent, even though they are somewhat different.

image

The next example here has at least 3 different shades of green within a small area. We probably shouldn't have the "Site Kit is Connected" text green either to not confuse it with a hyperlink. The green checkmark should be the indicator of success here rather than the text color.
image

## WP Dashboard

The WP dashboard has always had alternate styles applied to keep our widget consistent with the rest of the dashboard rather than using all the same styles we use on SK screens. It looks like we're still doing this (note the blue links in the table), just inconsistently.

image
image

Note: I'm using the "Lawn" WP admin theme here. We don't make an effort to remain compatible much with alternate admin themes, but that's an issue now as well.


## Inconsistent use of separator below headers

Some elements have a line under the header while others do not. This is an issue today as well but may be worth addressing here.

image
image
image


## Left over color in feature tours

Looks like Joyride a color that was missed.

image

File: assets/sass/components/global/_googlesitekit-header.scss [hunk 10]
```diff
@@ -51,16 +63,8 @@
...
-
-			.googlesitekit-plugin--has-scrolled & {
-
-				@include shadow;
-			}
-		}
-
-		&.googlesitekit-header--has-navigation {
-			box-shadow: none;
+		body.site-kit_page_googlesitekit-settings & {
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/sass/components/global/_googlesitekit-header.scss]: Actually it's good you found this, because this should have been removed since we no longer have rounded corners in the header. 😁

File: assets/sass/config/_variables-mui3.scss (hunk 9 above)
Comment (asvinb) [reply assets/sass/config/_variables-mui3.scss]: Sounds good. Thanks for the context! Updated!

**Comment by asvinb** [comment]
Thanks @aaemnnosttv . I'll let @sebastianmantel provide feedback on the "Green Hues" section. Separators have been added in the Settings page and joyride colors have been fixed. About the WP Dashboard issue, I started taking a look but I feel this might be another ticket on its own since it's not straight forward. We are right now hardcoding the links to the blue https://github.com/google/site-kit-wp/blob/develop/assets/sass/components/wp-dashboard/_googlesitekit-wp-dashboard.scss#L36 and if we remove those styles, it'll fallback to the link color defined for `.googlesitekit-plugin a`. Trying to tweak that rule by specifying something like `.googlesitekit-plugin:not(#google_dashboard_widget) a` results in unwanted changes in the app since we are adding more specificity to that rule thus overwriting other rules we have for links. I suggest we create another ticket which we can spend time diving into the ideal solution. Let me know what you think 👍

**Comment by aaemnnosttv** [comment]
> Can you please give more context on the WP admin dashboard? how are you getting the green color? Is that coming from the plugin?

@sebastianmantel that's coming from the admin theme. In this case, I'm using a theme from the [Admin Color Schemes](https://wordpress.org/plugins/admin-color-schemes/) plugin from the core team to add additional themes. My concern wasn't about the color there but more about the overall contrast between the styling of the SK widget and other core widgets, specifically around the fonts and the sizes which stick out more than before. The black also looks darker than other text/headings.

To be clear, we don't fully support blending in with alternate admin color themes at the moment either, but I would say the current styles for our widget in the WP admin dashboard fit more naturally than shown above.

**Review by techanvil — CHANGES_REQUESTED** [review]
Hi @asvinb this is looking great. I have raised a few minor issues, please see what you think.

File: assets/sass/config/_variables.scss [hunk 11]
```diff
@@ -144,48 +144,48 @@ $wp-colors: (
...
 $c-wp-notice-alt: $c-deep-sky-blue;
 
 $c-background: $c-mercury;
-$c-background-subtle: $c-alabaster;
+$c-background-subtle: $c-surfaces-surface-1;
 $c-background-brand: $c-royal-blue;
 $c-background-notice: $c-golden-rod-yellow;
 $c-background-notice-warning: $c-golden-rod-yellow;
-$c-background-notice-info: $c-solitude;
+$c-background-notice-info: $c-surfaces-surface-2;
```
[the remark this answers is not in the document]
Comment (aaemnnosttv) [reply assets/sass/config/_variables.scss:175]: Also worth noting that the `Learn more` text is a different green than the arrow ⚠️ 

image

**Comment by aaemnnosttv** [comment]
Thanks @asvinb ! A few more things I noticed on the latest (cc: @sebastianmantel)

The notice link does not appear to be aligned vertically with the text:
https://google.github.io/site-kit-wp/storybook/pull/5314/?path=/story/global-notices--settings-info-notice-single-line
image

This green button on the error component looks a bit odd
image

[Another instance](https://google.github.io/site-kit-wp/storybook/pull/5314/?path=/story/adsense-module-settings--view-closed) where I think the text should no longer be green?
image

For the styling of ["recommended"](https://google.github.io/site-kit-wp/storybook/pull/5314/?path=/story/adsense-module--use-snippet-switch-toggled-on-default) I think this should probably become a badge (like "Experimental" or "US Only"). Maybe it could still have a green color of some kind but this looks inconsistent with other labels we have.
image

Regarding the `Switch` styling, the colors appear to be reversed with the primary color as the track rather than the main part of the component. Compare the above with the current which looks better IMO:
image


I just noticed the floating label for the Text Input is cutting off the last letter (at least in this instance). Is the sizing assuming a specific font face or size maybe? 
image

Another example of floating labels cut off
image

Vertical alignment of Badges look off in some places
image

File: assets/sass/config/_variables.scss (hunk 11 above)
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/sass/config/_variables.scss]: Thanks for catching that @techanvil . I've changed the background color to a lighter grey which passes AA. However we need to have a conversation with design on what's the best color to use here.

File: assets/js/components/GoogleChart.js [hunk 12]
```diff
@@ -224,6 +224,16 @@ export default function GoogleChart( props ) {
...
 
+	set( chartOptions, 'hAxis.textStyle.fontName', 'Google Sans Text' );
+	set( chartOptions, 'hAxis.textStyle.fontSize', 10 );
+	set( chartOptions, 'hAxis.textStyle.color', '#5f6561' );
+	set( chartOptions, 'vAxis.textStyle.fontName', 'Google Sans Text' );
+	set( chartOptions, 'vAxis.textStyle.color', '#5f6561' );
+	set( chartOptions, 'vAxis.textStyle.fontSize', 10 );
+	set( chartOptions, 'legend.textStyle.fontName', 'Google Sans Text' );
+	set( chartOptions, 'legend.textStyle.color', '#131418' );
+	set( chartOptions, 'legend.textStyle.fontSize', 12 );
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/js/components/GoogleChart.js]: Nice! Updated!

File: assets/sass/config/_variables.scss [hunk 13]
```diff
@@ -16,64 +16,64 @@
  * limitations under the License.
  */
 
+@import "variables-mui3";
+
 /*
  * Colors
  * Names from http://chir.ag/projects/name-that-color/
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/sass/config/_variables.scss:23]: @techanvil That is correct, color names are out of sync. What we did is just tweaking the existing colors to use the colors from the new color palette in case which are GM3 compatible. These colors are no longer meant to be used now, nor new color variables should be added since we should exclusively use colors in `assets/sass/config/_variables-mui3.scss` going forward.

**Comment by asvinb** [comment]
@aaemnnosttv I fixed the vertical alignment in the single line notice. As for the color changes, I suggest we have a deeper conversation with design to see how things can be tweaked since they are all from the Figma files. About the floating labels being cut off, we'll need to dig deeper to understand what's really happening unfortunately :(

**Comment by aaemnnosttv** [comment]
> About the floating labels being cut off, we'll need to dig deeper to understand what's really happening unfortunately :(

I tracked this down, it's happening in the MDC TextField. Unfortunately the multiplier used for converting the label length into a width is defined as a constant

https://github.com/material-components/material-components-web/blob/31e517cea3002785ad2936ebc6ef12317b9d4133/packages/mdc-textfield/foundation.ts#L247-L248
https://github.com/material-components/material-components-web/blob/31e517cea3002785ad2936ebc6ef12317b9d4133/packages/mdc-textfield/constants.ts#L53

The weird thing is that the other labels are not cut off and seem to be calculated just fine 🤔 The others are `Select`s rather than `TextField`s so maybe that's it. I think they use the same `FloatingLabel` though, so I'm confused 😅 

image

**Comment by aaemnnosttv** [comment]
I think this is good now. A quick summary of changes:

- `CTA` text colors for errors now use the proper color, and adjusted font styles per @sebastianmantel  
https://google.github.io/site-kit-wp/storybook/pull/5314/?path=/story/components-reporterror--default-report-error
- Fixed broken stories for Settings where proper route wasn't loaded, [added support for setting the initial `route` via `parameters.route`](https://github.com/google/site-kit-wp/pull/5314/commits/032a55fef7519a054dfea2831093b0daf4de7e67)
- Updated text color of module status text to match body  
https://google.github.io/site-kit-wp/storybook/pull/5314/?path=/story/settings--connected-services
- Updated colors for `Switch` components, particularly using the primary/darker color for the thumb and lighter for the track  
https://google.github.io/site-kit-wp/storybook/pull/5314/?path=/story/modules-analytics-settings-settingsedit--with-ga-4-and-ua-snippet
- Set proper color on floating labels for `TextField`s (this was missing before too) and fixed a problem with these same floating labels which caused the label text to be cut short (reason unknown)
https://google.github.io/site-kit-wp/storybook/pull/5314/?path=/story/tag-manager-module-setup-secondary-amp--set-up-a-new-container
- Updated AdSense "recommended" one-off element in its snippet switch to use a `Badge` with a new primary style for consistency  
https://google.github.io/site-kit-wp/storybook/pull/5314/?path=/story/adsense-module-settings--edit-open
- Updated WP dashboard styles to inherit fonts and most colors (we can't currently model this very well in Storybook because styles are loaded for all areas together)  
	<details>
		<summary>WP dashboard screenshots</summary>
	image
	image
	image
	</details>

**Comment by techanvil** [comment]
> white_check_mark This is good to go on my end, although for some reason VRT is not liking the global admin bar scenario. That is the only thing blocking this IMO.

Not sure what was causing this in the first instance, but when I ran this locally the relevant Admin Bar story was a match for the expected VRT output, so I have updated the reference images and the visual test suite is now passing.
